rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/common/util/RMListenerProxy.java

```diff
@@ -126,16 +131,19 @@ public boolean init(String url, Credentials credentials) throws RMException, Key
      */
     public BooleanWrapper disconnect() {
         try {
-            target.getMonitoring().removeRMEventListener();
+            getMonitoringOrFail().removeRMEventListener();
         } catch (Exception ignored) {
```

Comment:
I guess to understand why this exception is so inclusive. However, for debugging purpose, it would great to notify somewhere the failure, and why. A debug in logs looks adapted to me

Reply:
ok